pkg/cli/httpClient.go

```diff
@@ -0,0 +1,183 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	zotErrors "github.com/anuvu/zot/errors"
+)
+
+var httpClient *http.Client = createHTTPClient() //nolint: gochecknoglobals
+
+const httpTimeout = 5 * time.Second
+
+func createHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout: httpTimeout,
+	}
+}
+func makeGETRequest(url, username, password string, resultStructPointer interface{}) (http.Header, error) {
+	attemptRLimitSet()
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.SetBasicAuth(username, password)
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusUnauthorized {
+			return nil, zotErrors.ErrUnauthorizedAccess
+		}
+
+		bodyBytes, _ := ioutil.ReadAll(resp.Body)
+
+		return nil, errors.New(string(bodyBytes)) //nolint: goerr113
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(resultStructPointer); err != nil {
+		return nil, err
+	}
+
+	return resp.Header, nil
+}
+
+func attemptRLimitSet() {
+	var rLimit syscall.Rlimit
+
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+	if err != nil {
+		return
+	}
+
+	if rLimit.Cur == rLimit.Max {
+		return
+	}
+
+	rLimit.Cur = rLimit.Max
+
+	_ = syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+}
+
+func isURL(str string) bool {
```

Comment:
Are you using this method to validate the "add" cmd url?

Reply:
I am now. Thank you!